tsdb/engine/tsm1/array_cursor_optimization_bug_test.go

```diff
@@ -0,0 +1,185 @@
+package tsm1
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/influxdata/influxdb/tsdb"
+)
+
+// TestAscendingCursorDuplicateDataBug demonstrates the real bug that existed
+// in the ascending cursor optimization. The bug is encountered when cache data is merged with tsm data, specifically
+// in an earlier call to Next(), the cache must fill part but not all of the response buffer, which is 1000 points. TSM
+// data then fills the rest. The cache must not be exhausted. The following Next() sees that only TSM data is needed,
+// and incorrectly copies TSM data including the points from the previous Next() call. Additionally, cache data
+// needs to be older than most of the TSM data. There needs to be enough tsm data younger than the cache data to completely
+// fill the remaining space in the response buffer.
+func TestAscendingCursorDuplicateDataBug(t *testing.T) {
```

Comment:
AI helped me make this test more robust; but it also made it rather long with the setup and precondition verifications.